keyboards/zsa/voyager/matrix.c

```diff
@@ -0,0 +1,224 @@
+// Copyright 2023 ZSA Technology Labs, Inc <@zsa>
+// Copyright 2023 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "voyager.h"
+#include "i2c_master.h"
+
+extern matrix_row_t matrix[MATRIX_ROWS];     // debounced values
+extern matrix_row_t raw_matrix[MATRIX_ROWS]; // raw values
+static matrix_row_t raw_matrix_right[MATRIX_COLS];
+
+#define ROWS_PER_HAND (MATRIX_ROWS / 2)
+#ifndef VOYAGER_I2C_TIMEOUT
+#    define VOYAGER_I2C_TIMEOUT 100
+#endif
+
+extern bool mcp23018_leds[3];
```

Comment:
This declaration seems to be in conflict with the definition in https://github.com/qmk/qmk_firmware/blob/e97d1b88a2d33c5ebe2a0d983a4c7f9aea81d1de/keyboards/zsa/voyager/voyager.c#L9

It leads to the following error if LTO is manually enabled:
```
| keyboards/zsa/voyager/matrix.c:17:13: error: type of 'mcp23018_leds' does not match original declaration [-Werror=lto-type-mismatch]
 |    17 | extern bool mcp23018_leds[3];
 |       |             ^
 | keyboards/zsa/voyager/voyager.c:9:6: note: array types have different bounds
 |     9 | bool mcp23018_leds[2] = {0, 0};
 |       |      ^
 | keyboards/zsa/voyager/voyager.c:9:6: note: 'mcp23018_leds' was previously declared here
```

Reply:
odd that it doesn't error out in general.  But fixed.